### PR TITLE
fix: ensure * glob doesn't match across path separators

### PR DESCRIPTION
**Pull request by dallenpyrah**
## Problem

The `bash: true` option in picomatch makes `*` match like `**` (across path separators), which breaks directory structure patterns.

For example, `modules/*` was matching `modules/chat/stream` instead of just `modules/chat`.

## Solution

Changed to `bash: false` (the default) to ensure:
- `*` matches single path segment only
- `**` is required to match across directories
- Consistent with standard glob behavior

### Bug Fix
- `*` glob no longer matches across path separators

### Cross-Platform Support
- All paths are now normalized (Windows backslashes → forward slashes)
- Trailing slashes are normalized in both paths and patterns
- Works correctly on both Windows and Unix

### Performance
- Added matcher cache to avoid recompiling same patterns
- Added `clearMatcherCache()` for testing/cache invalidation

### Edge Cases Handled
- Empty patterns (only match empty string)
- Empty paths
- Paths with just `." or `.."
- Trailing/leading slashes
- Special characters in paths
- Negation patterns
- Extglob patterns
- Character classes

### Documentation
- Added `CHANGELOG.md` with migration guide
- Added Glob Patterns section to `README.md`
- Comprehensive JSDoc with examples for all functions

### Tests
- 42 matcher tests (up from 11)
- Regression test for the specific bug
- Windows path tests
- Edge case coverage
- Cache functionality tests

## Breaking Change

This is a breaking change. Patterns relying on `*` matching across `/` need to use `**` instead.

See CHANGELOG.md for migration guide.

### CODE REVIEW

File: src/core/matcher.ts [hunk 1]
```diff
@@ -3,15 +3,22 @@ import picomatch from "picomatch";
 
 export type Matcher = (path: string) => boolean;
 
+/**
+ * Picomatch options for path matching.
+ * - dot: true - match dotfiles (.gitignore, .env, etc.)
+ * - bash: false - ensure * doesn't match across path separators (use ** for that)
+ */
+const MATCH_OPTIONS = { dot: true, bash: false } as const;
```
[the remark this answers is not in the document]
Comment (dallenpyrah) [reply src/core/matcher.ts]: ✅ Fixed in commit 1c532c4

Basename-only patterns (without `/`) are now auto-expanded to match anywhere:

```
*.log    → **/*.log   (matches src/debug.log)
*.d.ts   → **/*.d.ts  (matches types/index.d.ts)
!*.js    → !**/*.js   (negation handled correctly)
```

NOT expanded:
- `src/*.ts` (has path separator)
- `package.json` (no glob characters)
- `**/*.ts` (already has `**`)

8 new tests added for basename expansion behavior.
